src/object/obj_ec.h

```diff
@@ -360,15 +360,22 @@ obj_ec_singv_cell_bytes(uint64_t rec_gsize, struct daos_oclass_attr *oca)
 /** Query local record size and needed padding for evenly distributed singv */
 static inline void
 obj_ec_singv_local_sz(uint64_t rec_gsize, struct daos_oclass_attr *oca,
-		      uint32_t tgt_idx, struct obj_ec_singv_local *loc)
+		      uint32_t tgt_idx, struct obj_ec_singv_local *loc,
+		      bool update)
 {
 	uint32_t	data_tgt_nr = obj_ec_data_tgt_nr(oca);
 	uint64_t	cell_size;
 
 	D_ASSERT(tgt_idx < obj_ec_tgt_nr(oca));
 
 	cell_size = obj_ec_singv_cell_bytes(rec_gsize, oca);
-	if (tgt_idx >= data_tgt_nr)
+	/* For update, the parity buffer is immediately following data buffer.
+	 * For fetch, fetching from parity shard only for EC recovery, in that
+	 * case it allocates enough buffer (obj_ec_singv_stripe_buf_size) and
+	 * to simplify data recovery (avoid data movement) the parity data's
+	 * location is aligned to cell size boundary.
+	 */
+	if (tgt_idx >= data_tgt_nr && update)
```

Comment:
Sorry, I think this is will complicate the logic, and I would rather to take care the alignment when it is necessary, i.e. when calculate the parity.
this will easily get lost when following the code later. IMHO.